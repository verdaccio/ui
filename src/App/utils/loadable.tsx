import React, { lazy, Suspense } from 'react';
export default (importCallback: any) => {
  const TargetComponent = lazy(importCallback);
  return (props: any) => (
    <Suspense fallback={null}>
      <TargetComponent {...props} />
    </Suspense>
  );
};
