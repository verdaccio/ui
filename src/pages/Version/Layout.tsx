import React, { FC, ReactElement } from 'react';
import Grid from '@material-ui/core/Grid';
import DetailContainer from '../../components/DetailContainer';
import DetailSidebar from '../../components/DetailSidebar';

function renderDetail(): ReactElement<HTMLElement> {
  return <DetailContainer />;
}

function renderSidebar(): ReactElement<HTMLElement> {
  return <DetailSidebar />;
}

const Layout: FC<{}> = () => {
  return (
    <Grid className={'container content'} container={true} data-testid="version-layout" spacing={0}>
      <Grid item={true} md={8} xs={12}>
        {renderDetail()}
      </Grid>
      <Grid item={true} md={4} xs={12}>
        {renderSidebar()}
      </Grid>
    </Grid>
  );
};

export { Layout };
