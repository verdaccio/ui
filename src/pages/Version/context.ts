import { createContext, Consumer, Provider } from 'react';

import { DetailContextProps, VersionPageConsumerProps } from './version-config';

export const DetailContext = createContext<Partial<DetailContextProps>>({});

export const DetailContextProvider: Provider<Partial<VersionPageConsumerProps>> = DetailContext.Provider;
export const DetailContextConsumer: Consumer<Partial<VersionPageConsumerProps>> = DetailContext.Consumer;
