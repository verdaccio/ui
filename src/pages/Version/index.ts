export { DetailContext, DetailContextConsumer, DetailContextProvider } from './context';
export { VersionPageConsumerProps, DetailContextProps } from './version-config';

export { default } from './Version';
