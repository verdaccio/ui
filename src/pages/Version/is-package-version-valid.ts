import { PackageMetaInterface } from '../../../types/packageMeta';

function isPackageVersionValid(packageMeta: Partial<PackageMetaInterface>, packageVersion?: string): boolean {
  if (!packageVersion || typeof packageVersion === 'undefined') {
    // if is undefined, that means versions does not exist, we continue
    return true;
  }

  if (packageMeta.versions) {
    return Object.keys(packageMeta.versions).includes(packageVersion);
  }

  return false;
}

export default isPackageVersionValid;
