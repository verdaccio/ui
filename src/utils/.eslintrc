{
    "rules": {
        "no-invalid-this": 0
    }
}