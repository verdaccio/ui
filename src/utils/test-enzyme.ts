import { mount, shallow } from 'enzyme';
import React from 'react';

import ThemeProvider from 'verdaccio-ui/design-tokens/ThemeProvider';

const shallowWithTheme = (element: React.ReactElement<any>, ...props: any[]) =>
  shallow(element, {
    wrappingComponent: ThemeProvider,
    ...props,
  });

const mountWithTheme = (element: React.ReactElement<any>, ...props: any[]): any =>
  mount(element, {
    wrappingComponent: ThemeProvider,
    ...props,
  });

export { mountWithTheme as mount, shallowWithTheme as shallow };
