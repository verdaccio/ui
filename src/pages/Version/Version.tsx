import React, { useContext } from 'react';

import Loading from 'verdaccio-ui/components/Loading';
import NotFound from 'verdaccio-ui/components/NotFound';

import { DetailContext } from './context';
import VersionLayout from './VersionLayout';

const Version: React.FC = () => {
  const detailContext = useContext(DetailContext);
  const { isLoading, hasNotBeenFound } = detailContext;

  if (isLoading) {
    return <Loading />;
  }

  if (hasNotBeenFound) {
    return <NotFound />;
  }

  return <VersionLayout />;
};

export default Version;
