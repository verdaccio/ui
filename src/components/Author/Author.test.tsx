import React from 'react';

import { mount } from 'verdaccio-ui/utils/test-enzyme';

import { DetailContext } from '../../pages/Version';

import Authors from './Author';

const withAuthorComponent = (packageMeta: React.ContextType<typeof DetailContext>['packageMeta']): JSX.Element => (
  <DetailContext.Provider value={{ packageMeta }}>
    <Authors />
  </DetailContext.Provider>
);

describe('<Author /> component', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  test('should render the component in default state', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        author: {
          name: 'verdaccio user',
          email: 'verdaccio.user@verdaccio.org',
          url: '',
          avatar: 'https://www.gravatar.com/avatar/000000',
        },
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = mount(withAuthorComponent(packageMeta));
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should render the component when there is no author information available', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = mount(withAuthorComponent(packageMeta));
    expect(wrapper.html()).toBeNull();
  });

  test('should render the component when there is no author email', () => {
    const packageMeta = {
      latest: {
        name: 'verdaccio',
        version: '4.0.0',
        author: {
          name: 'verdaccio user',
          url: '',
          avatar: 'https://www.gravatar.com/avatar/000000',
        },
        dist: { fileCount: 0, unpackedSize: 0 },
      },
      _uplinks: {},
    };

    const wrapper = mount(withAuthorComponent(packageMeta));
    expect(wrapper.html()).toMatchSnapshot();
  });
});
