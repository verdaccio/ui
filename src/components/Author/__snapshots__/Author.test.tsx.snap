// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Author /> component should render the component in default state 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-weight: 700;
}

.emotion-2 {
  padding: 0;
}

.emotion-2:hover {
  background-color: transparent;
}

.emotion-4 {
  padding: 0 10px;
  margin: 0;
}

<body>
    <div>
      <ul
        class="MuiList-root MuiList-padding MuiList-subheader"
      >
        <span
          class="MuiTypography-root emotion-0 emotion-1 MuiTypography-subtitle1"
        >
          Author
        </span>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root emotion-2 emotion-3 MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <a
            href="mailto:verdaccio.user@verdaccio.org?subject=verdaccio@4.0.0"
            target="_top"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circle"
            >
              <img
                alt="verdaccio user"
                class="MuiAvatar-img"
                src="https://www.gravatar.com/avatar/000000"
              />
            </div>
          </a>
          <div
            class="MuiListItemText-root emotion-4 emotion-5"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              verdaccio user
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </body>,
  "container": .emotion-0 {
  font-weight: 700;
}

.emotion-2 {
  padding: 0;
}

.emotion-2:hover {
  background-color: transparent;
}

.emotion-4 {
  padding: 0 10px;
  margin: 0;
}

<div>
    <ul
      class="MuiList-root MuiList-padding MuiList-subheader"
    >
      <span
        class="MuiTypography-root emotion-0 emotion-1 MuiTypography-subtitle1"
      >
        Author
      </span>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root emotion-2 emotion-3 MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <a
          href="mailto:verdaccio.user@verdaccio.org?subject=verdaccio@4.0.0"
          target="_top"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="verdaccio user"
              class="MuiAvatar-img"
              src="https://www.gravatar.com/avatar/000000"
            />
          </div>
        </a>
        <div
          class="MuiListItemText-root emotion-4 emotion-5"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            verdaccio user
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<Author /> component should render the component when there is no author email 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  font-weight: 700;
}

.emotion-2 {
  padding: 0;
}

.emotion-2:hover {
  background-color: transparent;
}

.emotion-4 {
  padding: 0 10px;
  margin: 0;
}

<body>
    <div>
      <ul
        class="MuiList-root MuiList-padding MuiList-subheader"
      >
        <span
          class="MuiTypography-root emotion-0 emotion-1 MuiTypography-subtitle1"
        >
          Author
        </span>
        <div
          aria-disabled="false"
          class="MuiButtonBase-root MuiListItem-root emotion-2 emotion-3 MuiListItem-gutters MuiListItem-button"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiAvatar-root MuiAvatar-circle"
          >
            <img
              alt="verdaccio user"
              class="MuiAvatar-img"
              src="https://www.gravatar.com/avatar/000000"
            />
          </div>
          <div
            class="MuiListItemText-root emotion-4 emotion-5"
          >
            <span
              class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
            >
              verdaccio user
            </span>
          </div>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
  </body>,
  "container": .emotion-0 {
  font-weight: 700;
}

.emotion-2 {
  padding: 0;
}

.emotion-2:hover {
  background-color: transparent;
}

.emotion-4 {
  padding: 0 10px;
  margin: 0;
}

<div>
    <ul
      class="MuiList-root MuiList-padding MuiList-subheader"
    >
      <span
        class="MuiTypography-root emotion-0 emotion-1 MuiTypography-subtitle1"
      >
        Author
      </span>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root emotion-2 emotion-3 MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle"
        >
          <img
            alt="verdaccio user"
            class="MuiAvatar-img"
            src="https://www.gravatar.com/avatar/000000"
          />
        </div>
        <div
          class="MuiListItemText-root emotion-4 emotion-5"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
          >
            verdaccio user
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
